src/messaging/ExchangeContext.cpp

```diff
@@ -238,6 +219,22 @@ ExchangeContext::ExchangeContext(ExchangeManager * em, uint16_t ExchangeId, Secu
     mFlags.Set(Flags::kFlagInitiator, Initiator);
     mDelegate = delegate;
 
+    ExchangeMessageDispatch * dispatch = nullptr;
+    if (delegate != nullptr)
+    {
+        // If there's a delegate, get the dispatch object from it.
+        dispatch = delegate->GetMessageDispatch(em->GetReliableMessageMgr(), em->GetSessionMgr());
+    }
+
+    // If the delegate is not provided, or the delegate hasn't provided a dispatch object, let's use
```

Comment:
having this an 'else' branch may be clearer

Reply:
That would have different semantics, no?  This is meant to handle the (allowed) case when `delegate->GetMessageDispatch()` returns null.

Reply:
I've restructured the code to convey the flow better.

Reply:
As long as the codesize is the same, with the now-duplicated "get the default" code...